sycl/source/detail/bindless_images.cpp

```diff
@@ -546,6 +546,37 @@ __SYCL_EXPORT void release_external_memory(external_mem extMem,
                           syclQueue.get_context());
 }
 
+__SYCL_EXPORT void
+unmap_external_linear_memory(void *mappedLinearRegion,
+                             const sycl::device &syclDevice,
+                             const sycl::context &syclContext) {
+  auto [urDevice, urCtx, Adapter] = get_ur_handles(syclDevice, syclContext);
+
+  Adapter->call<
+      sycl::errc::invalid,
+      sycl::detail::UrApiKind::urBindlessImagesFreeMappedLinearMemoryExp>(
+      urCtx, urDevice, mappedLinearRegion);
+}
+
+__SYCL_EXPORT void unmap_external_linear_memory(void *mappedLinearRegion,
+                                                const sycl::queue &syclQueue) {
+  unmap_external_linear_memory(mappedLinearRegion, syclQueue.get_device(),
+                               syclQueue.get_context());
+}
+
+__SYCL_EXPORT void unmap_external_image_memory(
+    image_mem_handle mappedImageMem, image_type imageType,
+    const sycl::device &syclDevice, const sycl::context &syclContext) {
+  free_image_mem(mappedImageMem, imageType, syclDevice, syclContext);
+}
+
+__SYCL_EXPORT void unmap_external_image_memory(image_mem_handle mappedImageMem,
+                                               image_type imageType,
+                                               const sycl::queue &syclQueue) {
+  unmap_external_image_memory(mappedImageMem, imageType, syclQueue.get_device(),
+                              syclQueue.get_context());
+}
```

Comment:
Are you sure this needs to be exported and not inlined on the user side?

Reply:
You're right. The functions taking a `sycl::queue` can be marked as `inline` in the `bindless_images.hpp` header and just call the variants taking `sycl::device` and `sycl::context` - instead of being marked as `__SYCL_EXPORT`.

I've amended the functions introduced in this PR which take a `sycl::queue` to be `inline` in the `bindless_images.hpp` header, and added definitions in that header that point to the variants taking the SYCL device and context which contain the actual definitions.

This should also apply to all of our other APIs, but I think this would be a refactoring change that should belong in a separate PR. We will track this issue internally.

Reply:
Well, I can have arguments for either approach, that's why I didn't ask for the change :)

Reply:
I feel that having the `sycl::queue` variants being `inline` should be an improvement. I'm not sure if it will have a large impact on compilation time, but at least the number of exported symbols will be reduced.

If you have a better intuition of the impact this might have on compilation time let me know, then we could expedite the change to the other Bindless Images API.